File: 13-02-2026/problem_5_substring_no_repeat/buggy/substring_no_repeat.cpp
// ============================================
#include <bits/stdc++.h>
using namespace std;

int main() {
    string s;
    cin >> s;
    vector<int> st;
    st.push_back(0);
    int best = 0;
    
    for (int i = 0; i < (int)s.size(); i++) {
        if (s[i] == ')') {
            st.push_back(i);
        } else {
            st.pop_back();
            if (st.empty()) {
                st.push_back(i - 1);
            }
            best = max(best, i - st.back());
        }
    }
    
    cout << best << "\n";
}


// =============================================
// PROBLEM 2: LONGEST SUBSTRING NO REPEAT - BUGGY C++

File: 13-02-2026/problem_5_substring_no_repeat/buggy/substring_no_repeat.py
# =============================================
# PROBLEM 2: LONGEST SUBSTRING NO REPEAT - BUGGY PYTHON
# =============================================
import sys

s = sys.stdin.readline().strip()
last = {}
l = 0
best = 0

for r, ch in enumerate(s):
    if ch in last and last[ch] > l:
        l = last[ch]
    last[ch] = r
    best = max(best, r - l)

print(best)
